Modules/_zstd/_zstdmodule.c

```diff
@@ -682,6 +682,9 @@ do {                                                                         \
     ADD_INT_CONST_TO_TYPE(mod_state->ZstdCompressor_type,
                           "FLUSH_FRAME", ZSTD_e_end);
 
+    /* Make ZstdCompressor immutable */
+    PyType_Freeze(mod_state->ZstdCompressor_type);
```

Comment:
Why not changing `zstd_compressor_type_spec` in `compressor.c`?

Reply:
Importing `_zstd` would raise `TypeError: cannot set 'CONTINUE' attribute of immutable type '_zstd.ZstdCompressor'`, because `type_setattro()` first checks the immutable type flag.

As far as I can tell there's no mechanism to set an associated constant in `PyType_Spec` / `PyType_Slot[]`, but I might have missed something.

Reply:
> TypeError: cannot set 'CONTINUE' attribute of immutable type '_zstd.ZstdCompressor'

Ah I see. One way to do it is to set the constants when creating the type instead of adding them dynamically. If it's too bothersome, let's add a comment in `compressor.c` explaining why the flag is not set here.

Reply:
> set the constants when creating the type instead of adding them dynamically

Do you have a pointer to how we'd do this whilst still using `PyType_FromModuleAndSpec`?

Reply:
I think you can bypass the checks by directly accessing the `tp_dict` via `_PyType_GetDict` (which returns a borrowed reference) but I'm not entirely sure (while it's said to be unsafe to work with `tp_dict` using the dict C API, we do it internally for some types such as the recent InterpolationType object, but the latter is a static type).

I'm also unsure whether we can have a custom metaclass which would be responsible for setting the class variable in its `__init__`, but I don't remember whether we support this kind of extension (IIRC, `PyType_FromMetaclass` has some limitations).

@encukou or @vstinner may have a better idea on this one. If this becomes too complex, we can just keep your construction.

Reply:
In this case, PyType_Freeze() usage looks appropriate.

Reply:
AFAIK, `PyType_Freeze()` was added specially for such cases.

Reply:
Ok, so `PyType_Freeze` it is. However, let's add two comments: one explaining why we don't have the flag and one why we call `PyType_Freeze()` at this point (as they live in two different files)

Reply:
A brief comment would not hurt. But it's not strictly necessary either -- if someone tries to change something, they'll get an error and see why the code is the way it is.